aggregate.go

```diff
@@ -7,122 +7,109 @@ import (
 	"github.com/giantswarm/microerror"
 )
 
-// Aggregate merges capabilities based on dependencies version bundles within
-// the given capabilities define for their components.
-func Aggregate(capabilities []Capability) (Aggregation, error) {
-	err := ValidateCapabilities(capabilities).Validate()
-	if err != nil {
-		return Aggregation{}, microerror.Mask(err)
+// Aggregate merges bundles based on dependencies version bundles within
+// the given bundles define for their components.
+func Aggregate(bundles []Bundle) ([][]Bundle, error) {
+	if len(bundles) == 0 {
+		return nil, nil
 	}
 
-	var newAggregration Aggregation
-
-	if len(capabilities) == 0 {
-		return newAggregration, nil
-	}
+	var groupedBundles [][]Bundle
 
-	if len(capabilities) == 1 {
-		newAggregration.Capabilities = append(newAggregration.Capabilities, capabilities)
-		return newAggregration, nil
+	if len(bundles) == 1 {
+		groupedBundles = append(groupedBundles, bundles)
+		return groupedBundles, nil
 	}
 
-	for _, c1 := range capabilities {
-		for _, b1 := range c1.Bundles {
-			c1.Bundles = []Bundle{
-				b1,
-			}
-
-			newCapabilities := []Capability{
-				c1,
-			}
-
-			for _, c2 := range capabilities {
-				for _, b2 := range c2.Bundles {
-					c2.Bundles = []Bundle{
-						b2,
-					}
-
-					if reflect.DeepEqual(c1, c2) {
-						continue
-					}
-
-					if capabilitiesConflictWithDependencies(c1, c2) {
-						continue
-					}
-
-					if capabilitiesConflictWithDependencies(c2, c1) {
-						continue
-					}
-
-					if containsCapabitlityWithBundleName(newCapabilities, c2) {
-						continue
-					}
+	for _, b1 := range bundles {
+		newGroup := []Bundle{
+			b1,
+		}
 
-					newCapabilities = append(newCapabilities, c2)
-				}
+		for _, b2 := range bundles {
+			if reflect.DeepEqual(b1, b2) {
+				continue
 			}
 
-			sort.Sort(SortCapabilitiesByName(newCapabilities))
-			for _, c := range newCapabilities {
-				sort.Sort(SortBundlesByVersion(c.Bundles))
+			if bundlesConflictWithDependencies(b1, b2) {
+				continue
 			}
 
-			if containsAggregatedCapabilities(newAggregration.Capabilities, newCapabilities) {
+			if bundlesConflictWithDependencies(b2, b1) {
 				continue
 			}
 
-			if len(capabilities) != len(newCapabilities) {
+			if containsBundleByName(newGroup, b2) {
 				continue
 			}
 
-			newAggregration.Capabilities = append(newAggregration.Capabilities, newCapabilities)
+			newGroup = append(newGroup, b2)
 		}
+
+		sort.Sort(SortBundlesByVersion(newGroup))
+		sort.Sort(SortBundlesByName(newGroup))
```

Comment:
I think second sort should be stable (`sort.Stable`).

Reply:
Done.